services/api.py

```diff
@@ -809,6 +809,20 @@ def service_nodes_by_ancestors(service_node_ids):
                 bbox_geometry_filter = munigeo_api.build_bbox_filter(ref, val, 'geometry')
                 queryset = queryset.filter(Q(**bbox_filter) | Q(**bbox_geometry_filter))
 
+        if 'category' in filters:
+            services_and_service_nodes = filters.get('category', None).split(',')
+            service_ids = []
```

Comment:
I would maybe write

    service_ids = set()

and later

    service_ids.add(x)

To sanitize the input a little bit, in case it contains duplicates. But a minor thing.